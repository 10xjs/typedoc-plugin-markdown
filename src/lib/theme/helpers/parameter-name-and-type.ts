import { ParameterReflection } from 'typedoc';
import { type } from './type';

export function parameterNameAndType(this: ParameterReflection) {
  const md = [];
  md.push('â– ');
  md.push(`\`${this.flags.isRest ? '...' : ''} ${this.name}\``);
  if (this.type) {
    md.push(`: *${type.call(this.type)}*`);
  }
  return md.join('');
}
