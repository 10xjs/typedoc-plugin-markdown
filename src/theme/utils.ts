import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
