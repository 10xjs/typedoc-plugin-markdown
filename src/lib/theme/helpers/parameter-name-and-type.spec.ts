import { Fixture, getFixture, getReflectionByName } from '../../../test/test.utils';
import { parameterNameAndType } from './parameter-name-and-type';

describe(`parameterNameAndType helper`, () => {
  test(`should compile`, () => {
    const data = getReflectionByName(getFixture(Fixture.LiteralObject), 'objectLiteral');
    const result = parameterNameAndType.call(data);
    expect(result).toMatchSnapshot();
  });
});
