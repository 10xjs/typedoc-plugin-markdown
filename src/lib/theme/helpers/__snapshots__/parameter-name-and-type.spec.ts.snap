// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parameterNameAndType helper should compile 1`] = `"â–ª\`Const\`  **objectLiteral**: *[object Object]*"`;
